src/libraries/Common/src/Interop/Unix/System.Security.Cryptography.Native/Interop.OpenSsl.cs

```diff
@@ -220,6 +341,18 @@ internal static SafeSslHandle AllocateSslContext(SslProtocols protocols, SafeX50
                     throw;
                 }
             }
+            finally
+            {
+                if (innerContext != null)
+                {
+                    // We allocated new context and we want to cache
+                    if (!cacheSslContext || sslAuthenticationOptions.CertificateContext?.SslContexts == null ||
+                        !sslAuthenticationOptions.CertificateContext.SslContexts.TryAdd(sslAuthenticationOptions.EnabledSslProtocols, innerContext))
+                    {
+                        innerContext.Dispose();
```

Comment:
In the event that we're not caching, what happens here?  Seems like we'll have created an SSL_CTX, from that we created an SSL, then without doing anything yet we call dispose on the SSL_CTX handle, which might call SSL_shutdown.

I feel like I must be missing something.